src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -0,0 +1,378 @@
+     *
+     * @AdminSecurity("is_granted('create', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function generateMailsAction(Request $request)
+    {
+        $generateThemeMailsForm = $this->createForm(GenerateMailsType::class);
+        $generateThemeMailsForm->handleRequest($request);
+
+        if ($generateThemeMailsForm->isSubmitted()) {
+            if (!$generateThemeMailsForm->isValid()) {
+                $this->flashErrors($this->getFormErrorsForJS($generateThemeMailsForm));
+
+                return $this->redirectToRoute('admin_mail_theme_index');
+            }
+
+            $data = $generateThemeMailsForm->getData();
+            try {
+                /** @var GenerateThemeMailsCommand $generateCommand */
+                $generateCommand = new GenerateThemeMailsCommand(
+                    $data['mailTheme'],
+                    $data['language'],
+                    $data['overwrite']
+                );
+                //Overwrite theme folder if selected
+                if (!empty($data['theme'])) {
+                    $themeFolder = $this->getParameter('themes_dir') . '/' . $data['theme'];
+                    $generateCommand
+                        ->setCoreMailsFolder($themeFolder . '/mails')
+                        ->setModulesMailFolder($themeFolder . '/modules')
+                    ;
+                }
+
+                /** @var CommandBusInterface $commandBus */
+                $commandBus = $this->get('prestashop.core.command_bus');
+                $commandBus->handle($generateCommand);
+
+                if ($data['overwrite']) {
+                    $this->addFlash(
+                        'success',
+                        $this->trans(
+                            'Successfully overrode mail templates for theme %s with locale %s',
+                            'Admin.Notifications.Success',
+                            [
+                                $data['mailTheme'],
+                                $data['language'],
+                            ]
+                        )
+                    );
+                } else {
+                    $this->addFlash(
+                        'success',
+                        $this->trans(
+                            'Successfully generated mail templates for theme %s with locale %s',
+                            'Admin.Notifications.Success',
+                            [
+                                $data['mailTheme'],
+                                $data['language'],
+                            ]
+                        )
+                    );
+                }
+            } catch (CoreException $e) {
+                $this->flashErrors([
+                    $this->trans(
+                        sprintf(
+                            'Could not generate mail templates for theme %s with locale %s',
```

Comment:
Perhaps _Cannot generate email templates for theme %s with locale %s_ to harmonize all wordings?

Reply:
Yes perfect